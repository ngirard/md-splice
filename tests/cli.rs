use assert_cmd::Command;
use insta::assert_snapshot;

fn cmd() -> Command {
    Command::cargo_bin(env!("CARGO_PKG_NAME")).unwrap()
}

#[test]
fn test_i1_version_flag() {
    let output = cmd().arg("--version").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_version", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag() {
    let output = cmd().arg("--help").output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_insert() {
    let output = cmd().args(["insert", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_insert", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}

#[test]
fn test_i1_help_flag_replace() {
    let output = cmd().args(["replace", "--help"]).output().unwrap();
    let stdout = String::from_utf8(output.stdout).unwrap();

    assert_snapshot!("i1_help_replace", stdout, {
        ".version" => "md-splice [VERSION]"
    });
}
