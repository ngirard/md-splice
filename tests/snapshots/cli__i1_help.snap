---
source: tests/cli.rs
expression: redact_version(&stdout)
---
Splice and modify Markdown files with AST-level precision.

Usage: md-splice [OPTIONS] <COMMAND>

Commands:
  insert   Insert new Markdown content at a specified position
  replace  Replace a Markdown node with new content
  help     Print this message or the help of the given subcommand(s)

Options:
  -f, --file <FILE_PATH>      The Markdown file to modify
  -o, --output <OUTPUT_PATH>  Write the output to a new file instead of modifying the original
  -h, --help                  Print help
  -V, --version               Print version
